src/_pytest/runner.py

```diff
@@ -223,13 +223,26 @@ def pytest_report_teststatus(report: BaseReport) -> Optional[Tuple[str, str, str
 def call_and_report(
     item: Item, when: Literal["setup", "call", "teardown"], log: bool = True, **kwds
```

Comment:
I believe we can further reduce `item.ihook` calls by also passing `ihook` as parameter? Then the caller (`runtestprotocol`) can get the `ihook` only once.

Reply:
I don't like passing `item.ihook` along with `item`, feels wrong...

Reply:
I can see that, OK then. 👍

Reply:
I agree that it feels wrong, still if it brings measurable perf improvements it may be worth the structure

Reply:
I might do it next time I look at a profile :)